zipline/utils/events.py

```diff
@@ -443,6 +443,18 @@ def calculate_start_and_end(self, dt, env):
             )
         )
 
+        # If after applying the offset to the start/end day of the week, we get
+        # day in a different week, skip this week and go on to the next
+        while next_trading_day.date().isocalendar()[1] != \
```

Comment:
Ah, right! week number is what you're looking for.  

Since a datetime has all the information of a date, you actually don't need to call `date()` throughout here.